docs/source/spec/http-protocol-compliance-tests.rst

```diff
@@ -0,0 +1,534 @@
+      - [string]
+      - A list of query string parameter names that must not appear in the
+        serialized HTTP request.
+
+        Each value MUST appear in the format in which it is sent over the
+        wire; if a key needs to be percent-encoded, then it MUST appear
+        percent-encoded in this list.
+    * - requireQueryParams
+      - [string]
+      - A list of query string parameter names that MUST appear in the
+        serialized request URI, but no assertion is made on the value.
+
+        Each value MUST appear in the format in which it is sent over the
+        wire; if a key needs to be percent-encoded, then it MUST appear
+        percent-encoded in this list.
+    * - headers
+      - Map<String, String>
+      - A map of expected HTTP headers. Each key represents a header field
+        name and each value represents the expected header value. An HTTP
+        request is not in compliance with the protocol if any listed header
+        is missing from the serialized request or if the expected header
+        value differs from the serialized request value.
+
+        ``headers`` applies no constraints on additional headers.
+    * - forbidHeaders
+      - [string]
+      - A list of header field names that must not appear in the serialized
+        HTTP request.
+    * - requireHeaders
+      - [string]
+      - A list of header field names that must appear in the serialized
+        HTTP message, but no assertion is made on the value. Headers listed
+        in ``headers`` do not need to appear in this list.
+    * - body
+      - string
+      - The expected HTTP message body. If no response body is defined,
+        then no assertions are made about the body of the message.
+    * - bodyMediaType
+      - string
+      - The media type of the ``body``. This is used to help test runners
+        to parse and validate the expected data against generated data.
+        Binary media type formats require that the contents of ``body`` are
+        base64 encoded.
+    * - params
+      - document
+      - Defines the input parameters used to generate the HTTP request. These
+        parameters MUST be compatible with the input of the operation.
+    * - vendorParams
+      - document
+      - Defines vendor-specific parameters that are used to influence the
+        request. For example, some vendors might utilize environment
+        variables, configuration files on disk, or other means to influence
+        the serialization formats used by clients or servers.
+    * - documentation
+      - string
+      - A description of the test and what is being asserted defined in
+        CommonMark_.
+
+
+HTTP request example
+====================
+
+The following example defines a protocol compliance test for a JSON protocol
+that uses :ref:`HTTP binding traits <http-traits>`.
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        use smithy.test#httpRequestTests
+
+        @httpRequestTests([
```

Comment:
Nit: Can this be moved below the `@http` annotation for consistency with the other examples and the test case files?